spec/api/v2/resource/users_spec.rb

```diff
@@ -167,6 +167,25 @@
       activities = JSON.parse(response.body)
       expect(activities.first['id']).to be >= activities.last['id']
     end
+    it 'sorts user activities by result' do
+      4.times { create(:activity, user: test_user, result: 'succeed') }
+      get '/api/v2/resource/users/activity/all', params: { result: 'succeed' }, headers: auth_header
+      activities = JSON.parse(response.body)
+      expect(activities.length).to eq(4)
+    end
+    it 'sorts user activities by time range' do
+      4.times { create(:activity, user: test_user, result: 'succeed') }
+      get '/api/v2/resource/users/activity/all', params: { time_from: 5.hours.ago.to_i, time_to: 1.hours.ago.to_i }, headers: auth_header
+      activities = JSON.parse(response.body)
+      expect(activities.length).to eq(1)
+    end
+
+    it 'sorts user activities by time range' do
```

Comment:
i believe this tests doesn't show the use case, as you have only 4 activity in DB, and you fetch all 4.

Create 4 activity older then 5 hours, and 4 fresh. And check if filtering works correctly, dividing them